### PR TITLE
gephi: 0.9.6 -> 0.10.1

### PR DESCRIPTION
**Pull request by avdv**
###### Description of changes

This PR updates gephi to the latest stable version, and also adds the latest jogl version to javaPackages.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/java-modules/jogl/default.nix [hunk 1]
```diff
@@ -1,6 +1,79 @@
-{ lib, stdenv, fetchgit, ant, jdk8, git, xorg, udev, libGL, libGLU }:
+{ coreutils, lib, stdenv, fetchgit, ant, jdk8, jdk11, git, xorg, udev, libGL, libGLU, mesa, xmlstarlet }:
 
 {
+  jogl_2_4_0 =
+    let
+      version = "2.4.0";
+
+      gluegen-src = fetchgit {
```
Comment (drupol) [inline pkgs/development/java-modules/jogl/default.nix:8]: Looks like this derivation has more than one `src`, therefore, I would suggest to use `srcs`.

File: pkgs/development/java-modules/jogl/default.nix [hunk 2]
```diff
@@ -1,6 +1,79 @@
...
+        url = "git://jogamp.org/srv/scm/gluegen.git";
+        rev = "v${version}";
+        sha256 = "sha256-qQzq7v2vMFeia6gXaNHS3AbOp9HhDRgISp7P++CKErA=";
```
Comment (drupol) [inline pkgs/development/java-modules/jogl/default.nix:11]:
```suggestion
        hash = "sha256-qQzq7v2vMFeia6gXaNHS3AbOp9HhDRgISp7P++CKErA=";
```

File: pkgs/development/java-modules/jogl/default.nix [hunk 3]
```diff
@@ -1,6 +1,79 @@
...
+      };
+    in
+    stdenv.mkDerivation {
+      pname = "jogl";
+      inherit version;
+
+      src = fetchgit {
+        url = "git://jogamp.org/srv/scm/jogl.git";
+        rev = "v${version}";
+        sha256 = "sha256-PHDq7uFEQfJ2P0eXPUi0DGFR1ob/n5a68otgzpFnfzQ=";
```
Comment (drupol) [inline pkgs/development/java-modules/jogl/default.nix:22]:
```suggestion
        hash = "sha256-PHDq7uFEQfJ2P0eXPUi0DGFR1ob/n5a68otgzpFnfzQ=";
```

**Review by drupol — CHANGES_REQUESTED** [review]

File: pkgs/development/java-modules/jogl/default.nix (hunk 1 above)
Comment (avdv) [reply pkgs/development/java-modules/jogl/default.nix:8]: Thank you, TIL.

File: pkgs/development/java-modules/jogl/default.nix (hunk 3 above)
Comment (avdv) [reply pkgs/development/java-modules/jogl/default.nix:17]: done.

**Comment by avdv** [comment]
@drupol I have made the requested changes, PTAL.

**Review by drupol — APPROVED** [review]
LGTM, thanks for fixing everything :)

**Pull request merged**